source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -366,6 +366,12 @@ absl::Status ProcessorState::handleTrailersResponse(const TrailersResponse& resp
 }
 
 void ProcessorState::clearRouteCache(const CommonResponse& common_response) {
+  // No need to clear route cache for response traffic to the downstream client.
+  if (trafficDirection() == envoy::config::core::v3::TrafficDirection::OUTBOUND) {
```

Comment:
I would go with the change in the API as it should not allow clearing route cache when processing response headers. Given that API is still alpha, I think this is a better choice when just coding this restriction in the filter.

Reply:
Thanks for the comments.  IIUC, you are recommending removing 'clear_route_cache' 
https://github.com/envoyproxy/envoy/blob/0ae464feb4b3d1655baed1087cc97339fc644712/api/envoy/service/ext_proc/v3/external_processor.proto#L291
from 'CommonResponse'
https://github.com/envoyproxy/envoy/blob/0ae464feb4b3d1655baed1087cc97339fc644712/api/envoy/service/ext_proc/v3/external_processor.proto#L245
And put it into a proto message which is ext_proc server response specific to downstream client request.

However, currently the entire 'ProcessingResponse',  sent back by the ext_proc server to Envoy,
https://github.com/envoyproxy/envoy/blob/0ae464feb4b3d1655baed1087cc97339fc644712/api/envoy/service/ext_proc/v3/external_processor.proto#L126

and all its underline fields, including CommonResponse, doesn't differentiate whether it is for the downstream request or upstream response.  I went through the code, and does not find an easy way to make such changes. Please suggest if you have some good ideas.  Alternatively, we can add some comments for this field to mark it as something like "for responding to downstream request only".

Reply:
I am fine with the current way to add a check there.
If you have strong opinion this should be done by have a no-op in the ProcessorState class, and adding an override implementation in the derived class DecodingProcessorState, please let me know. I will change.